src/pane.js

```diff
@@ -586,14 +592,27 @@ export class Pane extends Cell {
         }
     }
     copySelection() {
-        const lines = this.t.getSelection().split('\n'),
-            linesFormatted = lines.map(l => l.trimEnd())
-    
-        return Clipboard.write({string: linesFormatted.join('\n')})
+        if (!this.cmSelection)
+            return
+        const serialized = this.serializeAddon.serialize()
+        const lines = serialized.split('\n')
+            .slice(this.cmSelection.startRow, this.cmSelection.endRow + 1)
+            .map(line => line.trim())
+            .filter(line => line.length > 0)
+        if (lines.length == 0)
+            return
+        if (lines.length == 1)
+            lines[0] = lines[0].slice(this.cmSelection.startColumn, this.cmSelection.endColumn + 1)
+        else {
+            lines[0] = lines[0].slice(this.cmSelection.startColumn)
+            lines[lines.length - 1] = lines[lines.length - 1].slice(0, this.cmSelection.endColumn + 1)
+        }
+        console.log(lines.join('\n'))
+        return Clipboard.write({string: lines.join('\n')})
```

Comment:
It still copy garbage to the clipboard like `mw[3` . Take a look at how we handle pointer up and copy the selection to the Clipboard

Reply:
oh we do have `buffer.activeg.getLine`, oops

Reply:
When a library call like `Clipboard.write` doesn't work as expected I always start by searching T7 code base. Chances are we use it in other place.

Reply:
we were using it before, hopefully it works now